workspaces/diff-engine/cli/src/main.rs

```diff
@@ -55,115 +83,170 @@ fn main() {
     },
   };
 
-  let events = SpecEvent::from_file(spec_file_path)
-    .map_err(|err| match err {
-      errors::EventLoadingError::Io(err) => {
-        eprintln!("Could not read specification file: {}", err);
-        process::exit(1);
-      }
-      errors::EventLoadingError::Json(err) => {
-        eprintln!("Specification JSON file could not be parsed: {}", err);
-        process::exit(1);
-      }
-      _ => unreachable!("Specification file not currently serialized as any other but JSON"),
-    })
-    .unwrap();
-
-  let spec_projection = Arc::new(SpecProjection::from(events));
-
   let mut runtime_builder = tokio::runtime::Builder::new_multi_thread();
   runtime_builder.enable_all();
   if let Some(core_threads) = core_threads_count {
     runtime_builder.worker_threads(core_threads as usize);
   }
 
   let runtime = runtime_builder.build().unwrap();
+
   runtime.block_on(async {
-    let stdin = stdin(); // TODO: deal with std in never having been attached
+    let spec_events = match spec_path_type {
+      SpecPathType::FILE => streams::spec_events::from_file(&spec_path)
+        .await
+        .map_err(|err| match err {
+          errors::EventLoadingError::Io(err) => {
+            eprintln!("Could not read specification file: {}", err);
+            process::exit(1);
+          }
+          errors::EventLoadingError::Json(err) => {
+            eprintln!("Specification JSON file could not be parsed: {}", err);
+            process::exit(1);
+          }
+          _ => unreachable!("Specification file not currently serialized as any other but JSON"),
+        })
+        .unwrap(),
+      SpecPathType::DIR => {
+        let spec_chunk_events = streams::spec_chunks::from_api_dir(&spec_path)
+          .await
+          .expect("should be able to find spec event chunks in a folder");
+
+        streams::spec_events::from_spec_chunks(spec_chunk_events)
+          .await
+          .unwrap() // TODO: report on these errors in a more user-friendly way (like for single spec files)
+      }
+    };
```

Comment:
No matter whether we're diffing or assembling, reading the spec from the filesystem is the same. Now inside of main Tokio runtime future, as the reading in both cases is now done with Async I/O.